modules/git/repo_tag.go

```diff
@@ -187,14 +187,27 @@ func (repo *Repository) GetTag(name string) (*Tag, error) {
 }
 
 // GetTagInfos returns all tag infos of the repository.
-func (repo *Repository) GetTagInfos() ([]*Tag, error) {
+func (repo *Repository) GetTagInfos(page, pageSize int) ([]*Tag, error) {
 	// TODO this a slow implementation, makes one git command per tag
 	stdout, err := NewCommand("tag").RunInDir(repo.Path)
 	if err != nil {
 		return nil, err
 	}
 
 	tagNames := strings.Split(strings.TrimRight(stdout, "\n"), "\n")
+
+	if page != 0 {
+		skip := (page - 1) * pageSize
+		if skip >= len(tagNames) {
+			return nil, nil
+		}
+		tagNames = tagNames[skip:]
+		if len(tagNames) < pageSize {
+			pageSize = len(tagNames)
+		}
+		tagNames = tagNames[:pageSize]
```

Comment:
You could calculate length so that you could use: `tagNames = tagNames[skip:pageSize]`

Reply:
@lafriks this produce weared outcome: https://play.golang.org/p/n5OCMLM1U-O